### ISSUE TITLE
inconsistency in seedSmarts in FMCS between and GetSubstructureMatches

### ISSUE THREAD

**Issue by proteneer** [rdkit/rdkit#3886]
**Describe the bug**

Providing an initial seed smarts to substructure searching returns valid matches for `mol_a` and `mol_b`. However, using the same seedSmarts in the FMCS code for `[mol_a, mol_b]` results in a single atom match. 

**To Reproduce**

``` python
import rdkit
from rdkit import Chem
from rdkit.Chem import rdFMCS

mol_a = Chem.MolFromSmiles("[H]OC1([H])c2c(c(Oc3c([H])nc([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C(F)(F)F)C([H])([H])C1(F)F")
mol_b = Chem.MolFromSmiles("[H]OC1([H])c2c(c(Oc3c([H])c(F)c([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C([H])(F)F)C([H])([H])C1(F)F")

initialSeed = "[#8](~[*]1~[*]~[*]~[*]~[*]~[*]~1)~[*]2~[*]~[*]~[*]~[*]~[*]~2" # doesn't work

query_mol = Chem.MolFromSmarts(initialSeed)
substructs = mol_a.GetSubstructMatches(query_mol)
print(substructs)
substructs = mol_b.GetSubstructMatches(query_mol)
print(substructs)

res = rdFMCS.FindMCS(
    [mol_a, mol_b],
    seedSmarts=initialSeed
)

print(res.smartsString)
```
Prints
```
((5, 4, 3, 2, 16, 15, 14, 6, 7, 8, 9, 10, 13),)
((5, 4, 3, 2, 17, 16, 15, 6, 7, 8, 10, 11, 14),)
[#16]
```

**Expected behavior**

The described initialSeed is present in substructure matching but fails in FMCS (i've tried toggling on/off various ring matching combinations, to no avail). 

**Screenshots**

mol_a:
![Screenshot from 2021-03-04 12-32-17](https://user-images.githubusercontent.com/2280724/110005800-e28ac680-7ce6-11eb-9981-b85e9580c9c6.png)
mol_b:
![Screenshot from 2021-03-04 12-32-22](https://user-images.githubusercontent.com/2280724/110005806-e3235d00-7ce6-11eb-8465-ab1ae5d4edfb.png)
seed:
![Screenshot from 2021-03-04 12-32-56](https://user-images.githubusercontent.com/2280724/110005810-e4548a00-7ce6-11eb-905b-003a5273c213.png)

**Configuration (please complete the following information):**
 - RDKit version: 5faa9c393c1be7781e3430c5fbbebe955a2e20d4
 - OS: Ubuntu 18.04.2 LTS
 - Python version (if relevant): 3.7
 - Are you using conda? no, built directly from commit above.

**Comment by ptosco** [comment]
Hi Yutong, thanks for the reproducible. I'll look into fixing this.

**Comment by bp-kelley** [comment]
On my nightly this works so I think this is already fixed.

```
void testSeed() {
  std::vector<ROMOL_SPTR> mols = { "[H]OC1([H])c2c(c(Oc3c([H])nc([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C(F)(F)F)C([H])([H])C1(F)F"_smiles,
      "[H]OC1([H])c2c(c(Oc3c([H])c(F)c([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C([H])(F)F)C([H])([H])C1(F)F"_smiles};
    
    auto initialSeed = "[#8](~[*]1~[*]~[*]~[*]~[*]~[*]~1)~[*]2~[*]~[*]~[*]~[*]~[*]~2";
    auto query_mol = "[#8](~[*]1~[*]~[*]~[*]~[*]~[*]~1)~[*]2~[*]~[*]~[*]~[*]~[*]~2"_smarts;
    MCSResult res = findMCS(mols);
    TEST_ASSERT(res.NumAtoms > 1);
}
```

**Comment by bp-kelley** [comment]
SmartsString = "[#8]-[#6]1-[#6]2:[#6](-[#6]-[#6]-1(-[#9])-[#9]):[#6](-[#8]-[#6](:[#6]):[#6]:[#6](:[#6])-[#6]#[#7]):[#6]:[#6]:[#6]:2-[#16](=[#8])(=[#8])-[#6](-[#9])-[#9]"

**Comment by bp-kelley** [comment]
Working failing test

```
void testSeed() {
  std::vector<ROMOL_SPTR> mols = { "[H]OC1([H])c2c(c(Oc3c([H])nc([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C(F)(F)F)C([H])([H])C1(F)F"_smiles,
      "[H]OC1([H])c2c(c(Oc3c([H])c(F)c([H])c(C#N)c3[H])c([H])c([H])c2S(=O)(=O)C([H])(F)F)C([H])([H])C1(F)F"_smiles};
    
    auto initialSeed = "[#8](~[*]1~[*]~[*]~[*]~[*]~[*]~1)~[*]2~[*]~[*]~[*]~[*]~[*]~2";
    auto query_mol = "[#8](~[*]1~[*]~[*]~[*]~[*]~[*]~1)~[*]2~[*]~[*]~[*]~[*]~[*]~2"_smarts;
//TEST_ASSERT(mol_a.GetSubstructMatches(query_mol, uniquify=False).size() > 0);
//TEST_ASSERT(mol_b.GetSubstructMatches(query_mol, uniquify=False).size() > 0);
    MCSParameters p;
    p.InitialSeed = initialSeed;
    MCSResult res = findMCS(mols, &p);
    TEST_ASSERT(res.NumAtoms > 1);
    std::cout << res.SmartsString << std::endl;
}
```

**Comment by bp-kelley** [comment]
Simpler test case

```
mol_a = Chem.MolFromSmiles("c1cccnc1")
mol_b = Chem.MolFromSmiles("Fc1ccccc1")
seed = "N#Cc1ccc*c1"

query_mol = Chem.MolFromSmarts(seed)
substructs = mol_a.GetSubstructMatches(query_mol, uniquify=False)
print(substructs)
substructs = mol_b.GetSubstructMatches(query_mol, uniquify=False)
print(substructs)


res = rdFMCS.FindMCS(
    [mol_a, mol_b],
    seedSmarts=seed
)
print(res.smartsString)mol_a = Chem.MolFromSmiles("c1cccnc1")
mol_b = Chem.MolFromSmiles("Fc1ccccc1")
seed = "N#Cc1ccc*c1"

query_mol = Chem.MolFromSmarts(seed)
substructs = mol_a.GetSubstructMatches(query_mol, uniquify=False)
print(substructs)
substructs = mol_b.GetSubstructMatches(query_mol, uniquify=False)
print(substructs)


res = rdFMCS.FindMCS(
    [mol_a, mol_b],
    seedSmarts=seed
)
print(res.smartsString)
```

It appears that ring dummies are causing the issue.

**Comment by bp-kelley** [comment]
So the issue is the initial side, while matching, isn't an mcs with respect to the current atom typers.

**Comment by proteneer** [comment]
I think Brian's right on this one! I had realized that a check was done on the initial seed while growing the core. I'm going to close this issue and mark it as a user error.

**Comment by bp-kelley** [comment]
I think it’s a bug though, if the seed matches but fails Atom typing it should be detected, logged and ignored as a seed.

**Comment by ptosco** [comment]
@bp-kelley I'll fix this as soon as we are done with #3767 (that's my top priority at the moment), and certainly before release.
